java/org/apache/catalina/authenticator/SSLAuthenticator.java

```diff
@@ -104,7 +104,7 @@ protected String getAuthMethod() {
     }
 
     @Override
-    protected boolean isPreemptiveAuthRequest(Request request) {
+    protected boolean isPreemptiveAuthPossible(Request request) {
```

Comment:
I wonder whether TLS allows preemptive auth at all.

Reply:
I'm not sure either. Browsers definitely want a challenge first. I just tried to mimic the current behaviour as close as possible - the code is unchanged from what was in `AuthenticatorBase` before.

Reply:
I have just read RFC 5246 and RFC 8466 and there is no single definition of a preemptive Client Cert auth. So the server has to send `CertificateRequest` first. Since HTTP authz checks come after TLS, I fail to understand the previous code. Maybe @markt-asf can enlighted us.

Reply:
RFC 8466 is relatively clear about client certs sent to the server:
```
4.4.2.  Certificate
[...]

   The client MUST send a Certificate message if and only if the server
   has requested client authentication via a CertificateRequest message
   (Section 4.3.2).
```
So while it may be technically possible to start by sending client certificates it is definitely not supported by the spec. I would opt for deactivation of this "feature".

Reply:
Exactly, preemptive is forbidden/does not exist. Make it false/remove it.